test/position-managers/Gas.t.sol

```diff
@@ -0,0 +1,320 @@
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+
+import {IPositionManager, Actions} from "../../src/interfaces/IPositionManager.sol";
+import {PositionManager} from "../../src/PositionManager.sol";
+import {LiquidityRange} from "../../src/types/LiquidityRange.sol";
+import {IMulticall} from "../../src/interfaces/IMulticall.sol";
+import {Planner} from "../utils/Planner.sol";
+import {PosmTestSetup} from "../shared/PosmTestSetup.sol";
+
+contract GasTest is Test, PosmTestSetup, GasSnapshot {
+    using FixedPointMathLib for uint256;
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+    using Planner for Planner.Plan;
+
+    PoolId poolId;
+    address alice;
+    uint256 alicePK;
+    address bob;
+    uint256 bobPK;
+
+    // expresses the fee as a wad (i.e. 3000 = 0.003e18 = 0.30%)
+    uint256 FEE_WAD;
+
+    LiquidityRange range;
+
+    function setUp() public {
+        (alice, alicePK) = makeAddrAndKey("ALICE");
+        (bob, bobPK) = makeAddrAndKey("BOB");
+
+        deployFreshManagerAndRouters();
+        deployMintAndApprove2Currencies();
+
+        (key, poolId) = initPool(currency0, currency1, IHooks(address(0)), 3000, SQRT_PRICE_1_1, ZERO_BYTES);
+        FEE_WAD = uint256(key.fee).mulDivDown(FixedPointMathLib.WAD, 1_000_000);
+
+        // Requires currency0 and currency1 to be set in base Deployers contract.
+        deployAndApprovePosm(manager);
+
+        // Give tokens to Alice and Bob.
+        seedBalance(alice);
+        seedBalance(bob);
+
+        // Approve posm for Alice and bob.
+        approvePosmFor(alice);
+        approvePosmFor(bob);
+
+        // define a reusable range
+        range = LiquidityRange({poolKey: key, tickLower: -300, tickUpper: 300});
+    }
+
+    function test_gas_mint() public {
+        Planner.Plan memory planner =
+            Planner.init().add(Actions.MINT, abi.encode(range, 10_000 ether, address(this), ZERO_BYTES));
+        bytes memory calls = planner.finalize(range.poolKey);
+        lpm.modifyLiquidities(calls, _deadline);
+        snapLastCall("PositionManager_mint");
+    }
+
+    function test_gas_mint_differentRanges() public {
+        // Explicitly mint to a new range on the same pool.
+        LiquidityRange memory bob_mint = LiquidityRange({poolKey: key, tickLower: 0, tickUpper: 60});
+        vm.startPrank(bob);
+        mint(bob_mint, 10_000 ether, address(bob), ZERO_BYTES);
+        vm.stopPrank();
+        // Mint to a diff range, diff user.
+        Planner.Plan memory planner =
+            Planner.init().add(Actions.MINT, abi.encode(range, 10_000 ether, address(alice), ZERO_BYTES));
+        bytes memory calls = planner.finalize(range.poolKey);
+        vm.prank(alice);
+        lpm.modifyLiquidities(calls, _deadline);
+        snapLastCall("PositionManager_mint_warmedPool_differentRange");
```

Comment:
Oh I just added this because the gas costs were different when I didn't "touch" a pool before the mint. Like compare this to just `PositionManager_mint`.. they are different. Im not really sure why I'm guessing its because the tokens are touched, and pool key is touched? But not totally sure.